sql/api/src/main/antlr4/org/apache/spark/sql/catalyst/parser/SqlBaseParser.g4

```diff
@@ -42,6 +42,28 @@ options { tokenVocab = SqlBaseLexer; }
   public boolean double_quoted_identifiers = false;
 }
 
+compoundOrSingleStatement
+    : singleStatement
+    | singleCompoundStatement
+    ;
+
+singleCompoundStatement
+    : beginEndCompoundBlock SEMICOLON? EOF
```

Comment:
```suggestion
    : beginEndCompoundBlock SEMICOLON* EOF
```

Reply:
Can you check [this](https://github.com/apache/spark/pull/46665#discussion_r1613389337) discussion in one of the previous comments?

We (together with Serge) decided to allow only one semi colon after each statement as it is more restrictive (and we can release that condition easily if needed).

So, I applied the same in this case as well. Does that make sense to you or you still think it should be changed?

Reply:
oh I see